Plugin.pm

```diff
@@ -357,6 +374,36 @@ sub trackInfoMenu {
 	};
 }
 
+sub _completed {
+	my ($client, $cb) = @_;
+	$cb->({
+		items => [{
+			type => 'text',
+			name => cstring($client, 'COMPLETE'),
+		}],
+	});
+}	
+
+sub addPlayingToFavorites {
+	my ($client, $cb, $args, $params) = @_;
+
+	my $id = Plugins::TIDAL::ProtocolHandler::getId($params->{url});
+	return _completed($client, $cb) unless $id;
+
+	Plugins::Deezer::Plugin::getAPIHandler($client)->updateFavorite( sub {
+		_completed($client, $cb);
+	}, 'add', 'tracks', $id );
+}
+
+sub addPlayingToPlaylist {
+	my ($client, $cb, $args, $params) = @_;
+
+	my $id = Plugins::Deezer::ProtocolHandler::getIdId($params->{url});
+	return _completed($client, $cb) unless $id;
+
+	addToPlaylist($client, $cb, { }, { id => $id }),
```

Comment:
what's `addToPlaylist()` supposed to do? It's not defined...

Reply:
Should this be `Plugins::TIDAL::InfoMenu::addToPlaylist()`?

Reply:
It is now, was added to the other PR